### PR TITLE
Allow observation files with a single variable in joint assimilation

### PR DESCRIPTION
**Pull request by mmazzolini**
Hi Esteban.. 

this is a very rough proposal for allowing observation files with a single observation. Feel free to discard it.

Cheers.
Marco.

PS: I debugged also this line https://github.com/ealonsogzl/MuSA/blob/76639db4ffaea7053860642d1589ac2fbeaa3c88/modules/internal_fns.py#L61 . 
I don't understand how that was working.. shouldn't observations have different column counting than df_state ? I don't get how that was working preciously.

### CODE REVIEW

File: modules/internal_fns.py [hunk 1]
```diff
@@ -183,37 +185,44 @@ def obs_array(dates_obs, lat_idx, lon_idx):
...
+                    if np.ma.is_masked(nc_value):
+                        nc_value = nc_value.filled(np.nan)
+                    else:
+                        nc_value = np.ma.getdata(nc_value)
+    
+                    tmp_error_storage.extend(nc_value)
+                else:
+    
+                    tmp_error_storage = [r_cov[cont]] * len(tmp_obs_storage)
+            except:
```
Comment (ealonsogzl) [inline modules/internal_fns.py]: please do not use bare excepts. Add the exceptions you expect to find. If there was a problem in a corrupt netcdf for example, we want MuSA to throw an exception.

File: modules/spatialMuSA.py [hunk 2]
```diff
@@ -603,10 +603,13 @@ def obs_mask():
...
-            nc_value = data_temp.variables[obs_var][:, :, :]
-            nc_value = nc_value.filled(np.nan)
-            tmp_storage.extend(nc_value)
-            data_temp.close()
+            try:
+                nc_value = data_temp.variables[obs_var][:, :, :]
+                nc_value = nc_value.filled(np.nan)
+                tmp_storage.extend(nc_value)
+                data_temp.close()
+            except:
```
Comment (ealonsogzl) [inline modules/spatialMuSA.py]: Same here, bare except

**Review by ealonsogzl — CHANGES_REQUESTED** [review]
Well seen that problem. It has not caused problems yet because no more than two observations have been assimilated since it was implemented. 
please resolve this bare except comments before merging.
Regards

**Comment by mmazzolini** [comment]
Hi @ealonsogzl 
Removed the try/except blocks and a couple of print statements..

However, if there are files with none of the obs_var_names present no exception is raised.
I guess it's possible to write that error in the obs_array function.

But we would need to invert the order of the for loops in the obs_array function:  it's necessary to check in the obs files both the obs_var before raising the error. 
This is not possible now if the first for loop is through the obs_var_names. We would need to loop through the files first and through the obs_var_names afterwards.

If you want I can try to implement & test in the future but for now I think this is acceptable. Let me know what you think.

Cheers,
Marco.

**Comment by ealonsogzl** [comment]
great, thank you very much Marco.
If there are no observations in any variable MuSA in a file should not throw an exception, if it did we would not be able to activate the spatial module in case of all observation times in a cell do not store any observations (like in your recent runs). The filters are ready to just keep the prior if there is nothing to assim.

For my part, I don't think I will use this capability, as with my routines I find it much more comfortable to prepare the observations as before, plus I find it less error prone. So feel free to propose here the changes that make your life easier :)

cheers

**Pull request merged**